### PR TITLE
ImplicitAccount transition semantic validation

### LINKED ISSUE #1664: ImplicitAccount transition semantic validation
- [x] https://github.com/iotaledger/iota-sdk/pull/1598/files#diff-a7e3b6fbffd8e007899d191008a0c1af128ab97fa83b9a2ad6fbdf4970909cdeR326
- [ ] https://github.com/iotaledger/iota-sdk/pull/1598/files#diff-9d298c39e1a196063ee5c85fcaef36422d32da60698d7343a489411fd173475aR462

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes https://github.com/iotaledger/iota-sdk/issues/1664

### CODE REVIEW

**Review by DaughterOfMars — APPROVED** [review]

File: sdk/src/types/block/semantic/mod.rs [hunk 1]
```diff
@@ -111,6 +113,14 @@ impl<'a> SemanticValidationContext<'a> {
                 Output::Delegation(output) => (output.amount(), None, output.unlock_conditions()),
             };
 
+            if unlock_conditions.addresses().any(Address::is_implicit_account_creation) {
+                if has_implicit_account_creation_address {
+                    return Ok(Some(TransactionFailureReason::SemanticValidationFailed));
```
Comment (Thoralf-M) [inline sdk/src/types/block/semantic/mod.rs:118]: This tx failure reason here is because there can be only one implicit account creation input?

Comment (thibault-martinez) [reply sdk/src/types/block/semantic/mod.rs:118]: Yes

Comment (Thoralf-M) [reply sdk/src/types/block/semantic/mod.rs:118]: I really don't like this SemanticValidationFailed :angry:

**Review by Thoralf-M — APPROVED** [review]

File: sdk/src/types/block/semantic/mod.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/types/block/semantic/mod.rs:118]: I have asked them to add a new variant btw, will change when they do

**Pull request merged**